src/npctalk.cpp

```diff
@@ -2912,6 +2930,89 @@ void talk_function::stop_guard( npc &p )
     p.guard_pos = npc::no_goal_point;
 }
 
+void talk_function::become_overseer( npc &p )
+{
+    if( !g->u.has_trait( trait_FACTION_CAMP_TUTORIAL ) ){
+        g->u.set_mutation( trait_FACTION_CAMP_TUTORIAL );
+        if( query_yn( _("Would you like to review the faction camp description?") ) ){
+            faction_camp_tutorial();
+        }
+    }
+
+    const point omt_pos = ms_to_omt_copy( g->m.getabs( p.posx(), p.posy() ) );
+    oter_id &omt_ref = overmap_buffer.ter( omt_pos.x, omt_pos.y, p.posz() );
+
+    if( omt_ref.id() != "field" ){
+        popup( _("You must build your camp in an empty field.") );
+        return;
+    }
+
+    std::vector<std::pair<std::string, tripoint>> om_region = om_building_region( p, 1 );
+    for( const auto &om_near : om_region ){
+        if ( om_near.first != "field" && om_near.first != "forest" && om_near.first != "forest_thick" &&
+                om_near.first != "forest_water" && om_near.first.find("river_") == std::string::npos ){
```

Comment:
Don't compare strings, this is risky. Especially don't ever use `string::find` to compare strings - this may randomly fail without any warning, since the suffix is added automatically and may change in the future.

For fields and forests, use the hardcoded values in `set_oter_ids()`, in `overmap_cpp`. They are all in the form of `ot_forest`.
For rivers, use `is_river( ter_id )`.